### PR TITLE
Bug fixes and use Process API for Smina calculator

### PR DESCRIPTION
**Pull request by mjrosengrant**
- Clean up job names in push-to-registry.yml
- Change No poses found message from exception to warning
- Add logs for most UI activity, for better insight into how things are used
- Fix bug in ligand dropdown when menu closed and reopened
- Use Process API for Smina Calculations (Autodock4 will be updated soon)

### CODE REVIEW

File: .github/workflows/push-to-registry.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ jobs:
       id: login-ecr
       uses: aws-actions/amazon-ecr-login@v1
 
-    - name: Build and Push to ECR public
+    - name: Push docking-autodock4 to ECR
```
Comment (ajm13) [inline .github/workflows/push-to-registry.yml:34]:
```suggestion
    - name: Build and Push to ECR public
```
Step name should either go back to what it was, or be "docking-smina". Autodock4 step is still "Build and Push to ECR public" so we should keep it consistent

File: nanome_docking/autodock4/calculations.py [hunk 2]
```diff
@@ -153,9 +154,15 @@ def _start_vina(self, receptor_file_pdbqt, ligand_file_pdbqt, num_modes=5, exhau
...
 
         nanome.util.Logs.message("Autodock4 calculation started.")
-        process = subprocess.Popen(args, cwd=self.temp_dir, stdout=subprocess.PIPE)
+        # process = subprocess.Popen(args, cwd=self.temp_dir, stdout=subprocess.PIPE)
         self.handle_loading_bar(process, 1)
-        process.wait()
+        self.loading_bar_counter = 0
+        p = Process(vina_binary, args, buffer_lines=False)
+        p.on_error = Logs.error
+        p.output_text = True
```
Comment (ajm13) [inline nanome_docking/autodock4/calculations.py:162]:
```suggestion
        p = Process(vina_binary, args, output_text=True, buffer_lines=False)
        p.on_error = Logs.error
```
Can put output_text as kwarg for Process

File: nanome_docking/autodock4/calculations.py [hunk 3]
```diff
@@ -153,9 +154,15 @@ def _start_vina(self, receptor_file_pdbqt, ligand_file_pdbqt, num_modes=5, exhau
...
+        p.on_output = partial(self.handle_loading_bar, ligand_count) 
+        exit_code = await p.start()
+        Logs.message('Autodock4 exit code: {}'.format(exit_code))
         return dock_results
 
     def handle_loading_bar(self, process, ligand_count):
```
Comment (ajm13) [inline nanome_docking/autodock4/calculations.py:168]: Changes from smina/calculations.py should be matched here (current handle_loading_bar for autodock4 looks like it wouldn't work)

File: nanome_docking/smina/calculations.py [hunk 4]
```diff
@@ -62,20 +61,26 @@ def run_smina(self, ligand_pdb, receptor_pdb, site_pdb, output_sdf, log_file,
             seed = '0'
             smina_args.extend(['--seed', seed])
 
-        cmd = [SMINA_PATH, *smina_args]
-        with subprocess.Popen(cmd, stdout=subprocess.PIPE) as process:
-            self.handle_loading_bar(process, ligand_count)
+        self.loading_bar_counter = 0
+        p = Process(SMINA_PATH, smina_args, buffer_lines=False)
+        p.on_error = Logs.error
+        p.output_text = True
```
Comment (ajm13) [inline nanome_docking/smina/calculations.py:67]:
```suggestion
        p = Process(SMINA_PATH, smina_args, output_text=True, buffer_lines=False)
        p.on_error = Logs.error
```

**Review by ajm13 — CHANGES_REQUESTED** [review]

File: .github/workflows/push-to-registry.yml (hunk 1 above)
Comment (mjrosengrant) [reply .github/workflows/push-to-registry.yml:34]: yeah, something got mixed up in a merge here

**Review by ajm13 — APPROVED** [review]

**Pull request merged**